File: Usuario/backend/src/main/java/com/signspeak/backend/dto/MensajeConversacionDTO.java
package com.signspeak.backend.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.signspeak.backend.model.MensajeConversacion;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class MensajeConversacionDTO {

    private Long idMensaje;
    private Long idConversacion;
    private MensajeConversacion.TipoMensaje tipoMensaje;
    private String contenidoTexto;
    private Long idSenaDetectada;
    private String palabraSena;
    private String nombreSena;  // ✅ AGREGADO - para setNombreSena()
    private BigDecimal confianzaDeteccion;
    private LocalDateTime timestamp;
    private LocalDateTime fechaHora;  // ✅ AGREGADO - para setFechaHora()
    private Boolean esCorrecto;
    private Map<String, Object> metadata;
}

File: Usuario/mobile/app/src/main/java/com/signspeak/data/remote/dto/conversacion/MensajeConversacionDTO.kt
package com.signspeak.data.remote.dto.conversacion

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class MensajeConversacionDTO(
    val idMensaje: Long,
    val idConversacion: Long,
    val tipoMensaje: TipoMensaje,
    val contenidoTexto: String?,
    val idSenaDetectada: Long?,
    val palabraSena: String?,
    val nombreSena: String?,
    val confianzaDeteccion: BigDecimal?,
    val timestamp: String?,
    val fechaHora: String?,
    val esCorrecto: Boolean?,
    @SerializedName("metadata")
    val metadata: Map<String, Any?>? = null
)

enum class TipoMensaje {
    @SerializedName("TEXTO_USUARIO")
    TEXTO_USUARIO,

    @SerializedName("SENA_DETECTADA")
    SENA_DETECTADA,

    @SerializedName("TRADUCCION_A_TEXTO")
    TRADUCCION_A_TEXTO,

    @SerializedName("SISTEMA")
    SISTEMA,

    @SerializedName("ERROR")
    ERROR
}